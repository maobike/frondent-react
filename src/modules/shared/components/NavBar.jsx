import React, { useState } from 'react';

import { NavLink } from 'react-router-dom';

import {
    Collapse,
    Navbar,
    NavbarToggler,
    NavbarBrand,
    Nav,
} from 'reactstrap';


function NavBar() {
    const [isOpen, setIsOpen] = useState(false);

    const toggle = () => setIsOpen(!isOpen);

    return (
        <>
            <Navbar color="dark" dark={true} expand="md">
                <NavbarBrand className='text-warning' data-testid='titleApp'>Test Corserva</NavbarBrand>
                <NavbarToggler onClick={toggle} />
                <Collapse isOpen={isOpen} navbar>
                    <Nav className="me-auto" navbar>
                        <NavLink
                            to="/"
                            className={({ isActive }) => `nav-item nav-link  ${isActive ? 'active' : ''}`}
                        >
                            Usuarios
                        </NavLink>
                        <NavLink
                            to="other"
                            className={({ isActive }) => `nav-item nav-link  ${isActive ? 'active' : ''}`}
                        >
                            Art√≠culos
                        </NavLink>
                    </Nav>
                </Collapse>
            </Navbar>
        </>
    );
}

export default NavBar;