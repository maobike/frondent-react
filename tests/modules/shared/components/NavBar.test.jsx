import React from 'react';
import { render } from '@testing-library/react';
import NavBar from './../../../../src/modules/shared/components/NavBar'; // AsegÃºrate de que la ruta sea correcta
import { BrowserRouter } from 'react-router-dom';

describe('Test in NavBar component', () => { 

    test('Navbar renders correctly', () => {
        const { container } = render(
            <BrowserRouter>
                <NavBar />
            </BrowserRouter>
        );
    
        expect( container).toMatchSnapshot();
    });

    test('should show title component', () => { 

        const { getByText, getByTestId } = render(
            <BrowserRouter>
                <NavBar />
            </BrowserRouter>
        );

        const title = 'Test Corserva';
        expect( getByText(title) ).toBeTruthy();
        expect( getByTestId('titleApp').innerHTML ).toContain(title)
    })
})