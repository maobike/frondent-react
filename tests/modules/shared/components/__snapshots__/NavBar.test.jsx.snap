// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in NavBar component Navbar renders correctly 1`] = `
<div>
  <nav
    class="navbar navbar-expand-md navbar-dark bg-dark"
  >
    <div
      class="container-fluid"
    >
      <a
        class="text-warning navbar-brand"
        data-testid="titleApp"
      >
        Test Corserva
      </a>
      <button
        aria-label="Toggle navigation"
        class="navbar-toggler"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="collapse navbar-collapse"
      >
        <ul
          class="me-auto navbar-nav"
        >
          <a
            aria-current="page"
            class="nav-item nav-link  active"
            href="/"
          >
            Usuarios
          </a>
          <a
            class="nav-item nav-link  "
            href="/other"
          >
            Art√≠culos
          </a>
        </ul>
      </div>
    </div>
  </nav>
</div>
`;
